##### Sanzhar Sengerbayev SE-2006 HW4

In [5]:
import pandas as pd 
from sklearn.model_selection import train_test_split
from sklearn.linear_model import LinearRegression
from sklearn.metrics import r2_score
%matplotlib inline

data = pd.read_csv("Magnum.csv")
months = ['Март','Апрель','Май','Июнь','Июль','Август','Сентябрь','Октябрь']
data['Ноябрь'] = (data['Октябрь'] + data['Октябрь'] - data[['Август', 'Сентябрь', 'Октябрь']].mean(axis = 1)).astype(int)
x = data[months].to_numpy()
y = data['Ноябрь'].to_numpy()
y = y.reshape(-1,1)
x_train, x_test, y_train, y_test = train_test_split(x, y, test_size = 0.15, random_state = 0)
model = LinearRegression().fit(x_train,y_train)
y_pred = model.predict(x_test)
pd.DataFrame(data={'Predicted value':y_pred.flatten(),'Actual Value':y_test.flatten()})
print("Intercept: ", model.intercept_)
print("Coeffs: ", model.coef_)
r2_score(y_test,y_pred)

Intercept:  [-0.37329704]
Coeffs:  [[ 0.00309505 -0.00838207  0.02370165 -0.01374103 -0.006581   -0.33836385
  -0.34161656  1.6818781 ]]


0.999999949722701

##### First, I add the modules I need and their methods. Splitting the data into train and test datasets, linear regression model, sklearn.metrics to calculating the R^2 value. Formula for actual price of the product is: Price(Nov) = Price(Oct) + Price(Oct) - Mean(Price(Aug-Oct)). Than using DataFrame class to compare predicted value with actual. As result calculating the coeffs, intercepts and R^2 which equal to 0.999999949722701